File: Portal/DTOs/Notes/NoteDto.cs
namespace Portal.DTOs.Notes;

public class NoteDto
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public Guid? ProjectId { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: portal_mobile/lib/features/projects/models/note_dto.dart
class NoteDto {
  final String id;
  final String title;
  final String content;
  final String projectId;
  final DateTime createdAt;

  NoteDto({
    required this.id,
    required this.title,
    required this.content,
    required this.projectId,
    required this.createdAt,
  });

  factory NoteDto.fromJson(Map<String, dynamic> json) {
    return NoteDto(
      id: json['id']?.toString() ?? '',
      title: json['title'] ?? '',
      content: json['content'] ?? '',
      projectId: json['projectId']?.toString() ?? '',
      createdAt: DateTime.tryParse(json['createdAt'] ?? '') ?? DateTime.now(),
    );
  }
}